index.js

```diff
@@ -1,37 +1,69 @@
 var webRTCSwarm = require('webrtc-swarm')
 var signalhub = require('signalhub')
+var inherits = require('inherits')
 var events = require('events')
 var discoverySwarm = require('discovery-swarm')
 var swarmDefaults = require('datland-swarm-defaults')
-var hasWebRTC = !!require('get-browser-rtc')()
+var rtc = require('get-browser-rtc')
 
 var DEFAULT_SIGNALHUB = 'https://signalhub.mafintosh.com'
 
-module.exports = function (archive, opts) {
-  var emitter = new events.EventEmitter()
+function HyperdriveSwarm (archive, opts) {
+  if (!(this instanceof HyperdriveSwarm)) return new HyperdriveSwarm(archive, opts)
+  var self = this
+
   if (!opts) opts = {}
-  var swarmKey = (opts.signalhubPrefix || 'dat-') + (archive.discoveryKey || archive.key).toString('hex')
-
-  if (hasWebRTC) {
-    var ws = webRTCSwarm(signalhub(swarmKey, opts.signalhub || DEFAULT_SIGNALHUB))
-    ws.on('peer', function (peer) {
-      emitter.emit('peer', peer)
-      peer.pipe(archive.replicate()).pipe(peer)
-    })
-  }
-
-  if (process.versions.node) {
-    var ds = discoverySwarm(swarmDefaults({
-      hash: false,
-      stream: function (peer) {
-        emitter.emit('peer', peer)
-        return archive.replicate()
-      }
-    }, opts))
-    ds.once('listening', function () {
-      ds.join(archive.discoveryKey)
-    })
-    ds.listen(0)
-  }
-  return emitter
+
+  self.swarmKey = (opts.signalhubPrefix || 'dat-') + archive.discoveryKey.toString('hex')
+  console.log(self.swarmKey)
+  self.signalhub = opts.signalhub || DEFAULT_SIGNALHUB
+  self.archive = archive
+  self.browser = null
+  self.node = null
+  self.opts = opts
+  if (!!rtc()) self._browser(self.swarmKey)
+  if (process.versions.node) self._node(self.swarmKey)
+
+  events.EventEmitter.call(this)
+}
+
+inherits(HyperdriveSwarm, events.EventEmitter)
+
+HyperdriveSwarm.prototype._browser = function (swarmKey) {
+  var self = this
+  self.browser = webRTCSwarm(signalhub(swarmKey, self.signalhub))
+  self.browser.on('peer', function (peer) {
+    self.emit('browser-connection', peer)
+    peer.pipe(self.archive.replicate()).pipe(peer)
```

Comment:
actually i think we should emit these as normal connection events. we can set the 2nd argument to `{type: 'webrtc'}` to separate the two if you really wanted to.